spec/queries/keywords_query_spec.rb

```diff
@@ -30,12 +30,66 @@
       end
 
       context 'given filter_params is nil' do
-        it 'returns all keywords' do
-          user = Fabricate(:user)
-          Fabricate.times(2, :keyword, user: user)
+        it 'returns Keyword scope' do
           keywords = described_class.new(Keyword, nil).call
 
-          expect(keywords.count).to eq 2
+          expect(keywords).to eq Keyword
+        end
+      end
+    end
+
+    context 'given the user search for result_urls' do
+      context 'given the user search for a word vpn and match at least 1' do
```

Comment:
```suggestion
      context 'given the user searches for a word vpn and match at least 1' do
```

Reply:
That said, a "query" object, does not care about the user... 🤔 
Should we use more "argument-driven" descriptions? Like:

```suggestion
      context 'given filters with the word `vpn` and match at least `1`' do
```

Reply:
Now this filter is "allowed" to be used by a bot, by a rake task or a background job if needed 😇